### PR TITLE
[v6r17] Using context managers where available

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

File: ConfigurationSystem/private/ConfigurationData.py [hunk 1]
```diff
@@ -332,13 +331,12 @@ def __backupCurrentConfiguration( self, backupName ):
...
     if os.path.isfile( configurationFile ):
       gLogger.info( "Making a backup of configuration in %s" % backupFile )
-      try:
-        zf = zipfile.ZipFile( backupFile, "w", zipfile.ZIP_DEFLATED )
-        zf.write( configurationFile, "%s.backup.%s" % ( os.path.split( configurationFile )[1], backupName ) )
-        zf.close()
-      except Exception:
-        gLogger.exception()
-        gLogger.error( "Cannot backup configuration data file", "file %s" % backupFile )
+      with zipfile.ZipFile( backupFile, "w", zipfile.ZIP_DEFLATED ) as zf:
```
Comment (JanEbbing) [inline ConfigurationSystem/private/ConfigurationData.py]: The `try` has to come before the `with` statement, the constructor can throw an exception.

(I can't find this in the [documentation](https://docs.python.org/2/library/zipfile.html) but a quick test shows this:
```
with zipfile.ZipFile( '/myzip.zip', "w", zipfile.ZIP_DEFLATED ) as zf:
  try:
     zf.write('eggs.txt')
   except Exception:
     print 'exception caught'
```
 
> Traceback (most recent call last):
>  File "<stdin>", line 1, in <module>
>  File "/afs/cern.ch/user/j/jebbing/dirac_home/Linux_x86_64_glibc-2.12/lib/python2.7/zipfile.py", line 756, in >__init__
>    self.fp = open(file, modeDict[mode])
>IOError: [Errno 13] Permission denied: '/myzip.zip' )

File: Resources/Catalog/PoolXMLFile.py [hunk 2]
```diff
@@ -113,15 +116,11 @@ def _getPoolCatalogs( directory = '' ):
...
-          gLogger.debug( 'Unpacking catalog XML file %s' % ( os.path.join( directory, fname ) ) )
-          tarFile = tarfile.open( os.path.join( directory, fname ), 'r' )
-          for member in tarFile.getmembers():
-            tarFile.extract( member, directory )
+        gLogger.debug( 'Unpacking catalog XML file %s' % ( os.path.join( directory, fname ) ) )
+        with tarfile.open( os.path.join( directory, fname ), 'r' ) as tf:
+          for member in tf.getmembers():
+            tf.extract( member, directory )
             poolCatalogList.append( os.path.join( directory, member.name ) )
-        except Exception, x :
```
Comment (JanEbbing) [inline Resources/Catalog/PoolXMLFile.py]: This now doesn't catch the exception any more, is this intentional?

Comment (fstagni) [inline Resources/Catalog/PoolXMLFile.py]: yes, this is intentional: catching the exception was just printing an error, but then leading to unexpected result. So, in this case is better to just let the exception propagate.

**Comment by fstagni** [comment]
Review anyone?

File: Core/Utilities/CFG.py [hunk 3]
```diff
@@ -878,17 +874,16 @@ def loadFromFile( self, fileName ):
     :param fileName: File name to load the contents from
     :return: This CFG
     """
-    if gZipEnabled and fileName.find( ".zip" ) == len( fileName ) - 4:
+    if fileName.find( ".zip" ) == len( fileName ) - 4:
```
Comment (chaen) [inline Core/Utilities/CFG.py]: fileName.endswith('.zip')  ?

Comment (sposs) [inline Core/Utilities/CFG.py]:
```
os.path.splitext(fileName)[1] == ".zip"
```

Comment (fstagni) [inline Core/Utilities/CFG.py]: `zipfile.is_zipfile(fileName)` seems even better to me, do you agree?

Comment (sposs) [inline Core/Utilities/CFG.py]: obviously +1

Comment (chaen) [inline Core/Utilities/CFG.py]: ah yes ! Cool ! I guess this also checks the integrity of the archive and so on ?

**Comment by atsareg** [comment]
Review OK

**Pull request merged**